### PR TITLE
Linear shade loss model - continuation of pr 1725

### LINKED ISSUE #1690: Add linear shade loss model for thin film
**Is your feature request related to a problem? Please describe.**
Related to https://github.com/pvlib/pvlib-python/issues/1689, it would be nice to have a linear self-shade power loss model. 

**Describe the solution you'd like**
Once a shaded fraction is calculated, power loss can be calculated by modifying equation 6 from [1] and assuming that power loss is linear (as is done in SAM for thin film [2]).

For a linear shade loss, shade-corrected power, `Psc`, can be related to estimated power with no shade, `Pns`, with 

`Psc = Pns * (1 – FS * (1 – DF))`

where `DF` is diffuse fraction (I've used a fixed value, i.e., 0.2, but it could be estimated more precisely in each timestep using satellite data, dirint or gti_dirint, etc.[3]).

This approach works well when using measured (unshaded) POA irradiance.

**Describe alternatives you've considered**
...

**Additional context**
I'm not sure where the equation for `Psc` came from. I first implemented it in 2021, and more recently thought that it came directly from Lorenzo 2011 [1], but now I don't think that's the case. And I think SAM simply reduces beam irradiance for the linear shade model and already has diffuse irradiance separately, so does not need to calculate a diffuse fraction. 

This method does not explicitly account for reductions in diffuse irradiance due to row to row shade. 

This was presented on at PVRW 2023 [4].

[1] E. Lorenzo, L Narvarte, and J Muñoz. 2011. "Tracking and back-tracking". Progress in
Photovoltaics: Research and Applications 19: 747-753. https://doi.org/10.1002/pip.1085
[2] Gilman, P. et al.., (2018) SAM Photovoltaic Model Technical Reference Update. 93 pp.; NREL/TP-6A20-67399 
[3] K. Anderson, W. Hobbs. Improved CdTe PLR Estimates: Self-Shading and Spectral Mismatch. NREL PVRW, 2022. https://www.osti.gov/biblio/1846944 
[4] I. Azad, W. Hobbs, "Improved PV expected energy modeling with a simple self-shading model". NREL PVRW 2023. (proceedings pending)

### PR DESCRIPTION
**Pull request by echedey-ls**
- [x] Closes #1690
 - [x] Substitutes #1725 (originally from @mikofski )
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Addresses the linear thin film module losses.

[Comment](https://github.com/pvlib/pvlib-python/pull/1962#issuecomment-1999555920) from @kandersolar  at #1962:
> > you are saying that we need a substantive reference that says linear power loss models are valid for thin film modules. Is that right?

> What I'd really like to reference is a paper that deals with specifically the equation implemented in this PR: power_loss_fraction = shaded_fraction1d * (1 - diffuse_fraction).

@mikofski do you have a reference for that or should I change to irradiance losses and a SAM reference?

Wait for #1962 to be merged before this one so the docstring makes sense.

### CODE REVIEW

**Comment by mikofski** [comment]
I found this by googling:
https://energy.sandia.gov/wp-content/gallery/uploads/26-Littmann-Tracking-and-Diffuse-Shading.pdf0_.pdf

It’s for series 4 but explains why linear shade for orientation perpendicular to cells

**Comment by echedey-ls** [comment]
> I found this by googling:
https://energy.sandia.gov/wp-content/gallery/uploads/26-Littmann-Tracking-and-Diffuse-Shading.pdf0_.pdf
It’s for series 4 but explains why linear shade for orientation perpendicular to cells

I don't really get to see where it explains the shade is perpendicular to the cells. I mean, I see the statement but not the reasoning.

I would prefer some other reference, although I'll stick to that one. I will update the PR by the end of the week.

**Comment by mikofski** [comment]
Maybe if you combine this reference with the [FirstSolar series 4 user guide](https://www.firstsolar.com/-/media/First-Solar/Technical-Documents/User-Guides/Series-4-Module-User-Guide-North-America.ashx?la=en). The point is the cells are long and thin and strung in series from side to side. The user guide explains that they should only be oriented so that the cells extend from bottom to top, not left to right or damage will over. There is also a series 6/7 user guide.

**Comment by echedey-ls** [comment]
Thanks for that insightful manual. I like how the docstring is turning out!

These references don't have the model equation nor address what the model modifies (power yield or irradiance). I think it doesn't matter as long as we find some other source that deals with the model eventually. Maybe some code snippet on the public domain under First Solar authorship?

File: pvlib/shading.py [hunk 1]
```diff
@@ -342,3 +342,61 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
     # Eq. (5); angle between sun's beam and surface
     theta_T = np.degrees(np.arctan2(sx_prime, sz_prime))
     return theta_T
+
+
+def linear_shade_loss(shaded_fraction, diffuse_fraction):
+    r"""
+    Fraction of power lost to linear shade loss applicable to monolithic thin
+    film modules like First Solar CdTe, where the shadow is perpendicular to
+    cell scribe lines.
```
Comment (AdamRJensen) [inline pvlib/shading.py:351]:
```suggestion
    Fraction of power lost to linear shade loss.
    
     Linear shade loss is applicable to monolithic thin film modules like
     First Solar CdTe, where the shadow is perpendicular to
    cell scribe lines.
```

File: pvlib/shading.py [hunk 2]
```diff
@@ -342,3 +342,61 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
...
+
+    .. versionadded:: 0.10.5
+
+    Parameters
+    ----------
+    shaded_fraction : numeric
+        The fraction of the collector width shaded by an adjacent row. A
+        value of 1 is completely shaded and zero is no shade.
```
Comment (AdamRJensen) [inline pvlib/shading.py:359]:
```suggestion
        value of 1 corresponds to completely shaded and zero is no shading.
```

File: pvlib/shading.py [hunk 3]
```diff
@@ -342,3 +342,61 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
...
+    diffuse_fraction : numeric
+        The ratio of diffuse plane of array (poa) irradiance to global poa.
+        A value of 1 is completely diffuse and zero is no diffuse.
```
Comment (AdamRJensen) [inline pvlib/shading.py:362]:
```suggestion
        The ratio of diffuse plane of array (POA) irradiance to global POA.
        A value of 1 corresponds to all incident irradiance being diffuse,
        and a value of zero corresponds to all irradiance being direct.
```

File: pvlib/shading.py [hunk 4]
```diff
@@ -342,3 +342,61 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
...
+    Returns
+    -------
+    linear_shade_loss : numeric
+        The fraction of power lost due to linear shading. A value of 1 is all
+        power lost and zero is no loss.
+
+    Notes
+    -----
+    The reasoning behind this loss model is described in sections 4.2 and 4.3
+    of [1]_, page 8 of the manual.
```
Comment (AdamRJensen) [inline pvlib/shading.py:373]:
```suggestion
    The reasoning behind this loss model is described in sections 4.2 and 4.3
    of [1]_.
```

File: pvlib/shading.py [hunk 5]
```diff
@@ -342,3 +342,61 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
...
+
+    The loss is calculated as:
+
+    .. math::
+
+        \text{loss} = \text{shaded\_fraction}\cdot(1-\text{diffuse\_fraction})
```
Comment (AdamRJensen) [inline pvlib/shading.py:379]:
```suggestion
        \text{loss} = \text{shaded_fraction}\cdot(1-\text{diffuse_fraction})
```
I don't think you need the backslashes in \text environment, at least they currently are rendered in the documentation.
image

File: pvlib/shading.py [hunk 6]
```diff
@@ -342,3 +342,61 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
...
+    >>> loss = shading.linear_shade_loss(sf, diffuse_fraction=0.2)
+    >>> P_no_shade = 100  # [kWdc]  DC output from modules
+    >>> P_linear_shade = P_no_shade * (1-loss)  # [kWdc] output after loss
+    75.51797783654133
+
+    References
+    ----------
+    .. [1] First Solar, “First Solar Series 4 Module.” [Online]. Available at: https://web.archive.org/web/20230512210902/https://www.firstsolar.com/-/media/First-Solar/Technical-Documents/User-Guides/Series-4-Module-User-Guide-North-America.ashx?la=en [accessed 2024-04-12]
+    .. [2] First Solar, “Diffuse Irradiance and Tracker Simulations,” May
+    01-02, 2013. http://web.archive.org/web/20170127080529/https://energy.sandia.gov/wp-content/gallery/uploads/26-Littmann-Tracking-and-Diffuse-Shading.pdf0_.pdf [accessed 2024-04-12]
```
Comment (AdamRJensen) [inline pvlib/shading.py:400]:
```suggestion
    .. [2] First Solar, “Diffuse Irradiance and Tracker Simulations,” May 01-02, 2013. http://web.archive.org/web/20170127080529/https://energy.sandia.gov/wp-content/gallery/uploads/26-Littmann-Tracking-and-Diffuse-Shading.pdf0_.pdf [accessed 2024-04-12]
```

Comment (echedey-ls) [reply pvlib/shading.py]: Thanks for all these suggestions, Adam.

**Comment by echedey-ls** [comment]
I have the feeling that the reference still is an issue. If you maintainers think so, do you think the same reference as #2063 and #2070 , equation (3), is a sustainable reference?

https://www.sciencedirect.com/science/article/abs/pii/S092702481000454X

Pro: it compares the results in Table 2 and Fig. 4.
Cons: it is compared for non-monolithic modules

**Comment by mikofski** [comment]
Hi @echedey-ls the reference isn’t really valid for firstsolar modules because firstsolar don’t use bypass diodes and the power loss isn’t exactly the same as the fraction of direct beam shaded, because there is still diffuse sky and ground. I think the best references are Kevin’s & Adam’s paper and the poster from PVPMC 2023 (https://pvpmc.sandia.gov/download/5381/?tmstv=1719027091). Saying The loss is linear is just industry jargon four saying there is no electrical mismatch due to shade, so the only liss is the linear decrease in direct beam.

**Comment by kandersolar** [comment]
> I have the feeling that the reference still is an issue.

For me, yes.  I don't think we have a reference that clearly describes the specific model implemented here.  I spent some time previously trying to locate a better reference, but was unsuccessful.  

Without a good reference, I suggest we close this PR.  It's easy enough to pick it back up again if/when a good reference becomes available in the future.

**Comment by mikofski** [comment]
I disagree. The first solar series 4, 6 & 7 manuals suffice. I will ask first Solar if they have something more definitive. 

Is the question:
1. What is the orientation of cells in a first solar module and how are they strung?
2. Is it, do first solar modules have bypass diodes?
3. Or is it, how the current of the cell is affected by light as it is progressively shaded from zero to 100%?

There are references available for all of these questions already. Please do not close this PR. My preference would be to figure out how to complete it or merge as is. I don’t understand the blocker, but I do know that I am seeing more first solar modules on terrain and it will likely continue due to concerns over domestic manufacturing and tariffs

**Comment by mikofski** [comment]
> what the model modifies (power yield or irradiance)

The equations affect direct beam irradiance only, so as whatever diffuse light before evaluating effect on power.

The reference for the equation is [the poster from PVPMC 2023](https://pvpmc.sandia.gov/download/5381/?tmstv=1719027091)

**Comment by echedey-ls** [comment]
Hey @mikofski , I plan on finishing development on all my PRs to pvlib for now unless straight interest is shown in certain proposals and I find the time to work on them. I will be closing this for these reasons, but feel free to copy this branch and continue development if you wish.

Now that I better understand pvlib and the PV context, my two cents to consider that poster not enough to include it are the following:
- Whether the losses factor is applied to power or irradiance:
  > The equations affect direct beam irradiance only

  but the poster states

  > (r2r) shade affects all cells uniformly and reduces
output linearly proportional to the fraction of shade

  I understand output is power output, since irradiance is not an output.

  So if the poster was to be used as a reference, I think that a matching statement between the implementation and it is essential.

- For either possibility, a reference equation would be nice, at least to double check the implementation and document all the inputs.
- Probably `pvlib.shading` should conform to the same API in terms of shading losses factor: irradiance or power losses, and current code matches the latter. So the equation is not as direct as the linear loss presented here deduced from the text.
- I would love to see a comparison of different models, à la https://www.sciencedirect.com/science/article/abs/pii/S092702481000454X , fig 4:
  
  image
  Thou I understand the underlying effect and it seems intuitive, I wonder if there are any non-linearities that may affect the power yield (more or less question n. 3), and even if there aren't, just a validation of the model.
  I expect references to require this validation stage, just to know how much difference I can assign to each modelling step.
 [as a non-expert on what your expertise is and how DNV works, would it be very difficult to obtain real data and model the losses from the shaded fraction?, maybe publish at arxiv or similar]

Again, feel free to pick up this work and continue from there - I can support you with reviews; and thanks for showing interest in improving these capabilities of pvlib, I learnt a lot through this journey.

**Comment by mikofski** [comment]
Ha! I found a reference!
>L. Ngan, C. Jepson, A. Blekicki, and A. Panchula, “[Increased Energy Production of First Solar Horizontal Single-Axis Tracking PV Systems without Backtracking](https://ieeexplore.ieee.org/document/6744267),” in IEEE PVSC, 2013, [DOI: 10.1109/PVSC.2013.6744267](https://doi.org/10.1109/PVSC.2013.6744267)

**Comment by mikofski** [comment]
maybe better to leave this closed and merge it into #1725
